src/client/components/vision2/camera/shaders/world_line.frag

```diff
@@ -0,0 +1,280 @@
+precision lowp float;
+
+#define M_PI 3.1415926535897932384626433832795
+
+// Lens/projection parameters
+uniform vec2 viewSize;
+uniform float focalLength;
+uniform int projection;
+uniform vec2 centre;
+uniform vec2 k;
+
+// Line parameters
+uniform vec3 axis;
+uniform vec3 start;
+uniform vec3 end;
+
+// Style
+uniform vec4 color;
+uniform float lineWidth;
+
+varying vec2 vUv;
+
+#define RECTILINEAR_PROJECTION 1
+#define EQUIDISTANT_PROJECTION 2
+#define EQUISOLID_PROJECTION 3
+
+// TODO(trent) these should be moved into a separate GLSL file once there is a decent #include system
+/**
+ * Takes an undistorted radial distance from the optical axis and computes and applies an inverse distortion
+ *
+ * @param r the radius to undistort
+ * @param k the undistortion coefficents
+ *
+ * @return an distortion radius
+ */
+float distort(float r, vec2 k) {
+  // Uses the math from the paper
+  // An Exact Formula for Calculating Inverse Radial Lens Distortions
+  // https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4934233/pdf/sensors-16-00807.pdf
+  // These terms have been stripped back to only include k1 and k2 and only uses the first 4 terms
+  // if more are needed in the future go and get them from the original paper
+  // TODO if performance ever becomes an issue, this can be precomputed for the same k values
```

Comment:
You might want to compute b1-4 on the javascript side and upload it as a uniform.